### PR TITLE
DAOS-10756 event: only thread private event owner changes status of e…

### PR DESCRIPTION
**Pull request by mchaarawi**
…vent

Change the progress on the thread private event for only the event owner
to change the status of the event.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/api/client_internal.h [hunk 1]
```diff
@@ -56,6 +56,8 @@ struct daos_event_private {
 	unsigned int		is_barrier:1;
 	/** flag to indicate whether to convert DER to errno */
 	unsigned int		is_errno:1;
+	/** flag to indicate whether even is thread private */
```
Comment (jolivier23) [inline src/client/api/client_internal.h:59]: s/even/event

File: src/client/api/event.c [hunk 2]
```diff
@@ -520,6 +520,13 @@ ev_progress_cb(void *arg)
 	if (evx->evx_nchild_running > 0)
 		return 0;
 
+	/*
+	 * If it's a thread private event but it's not owned by this thread, don't change the status
+	 * as there could be race with the thread owner.
+	 */
+	if (evx->is_th_private && !daos_event_is_priv(daos_evx2ev(evx)))
```
Comment (jolivier23) [inline src/client/api/event.c:527]: how does it know it's not owned by this thread?

File: src/client/api/client_internal.h (hunk 1 above)
Comment (mchaarawi) [reply src/client/api/client_internal.h:59]: ok if i get +1s and don't need to re-push, i will change it with skip-test pragma (but will need force landing). queue is about 30hrs for testing.

File: src/client/api/event.c (hunk 2 above)
Comment (mchaarawi) [reply src/client/api/event.c:527]: I do assume that:
static __thread daos_event_t	ev_thpriv;
would be unique for each thread

Comment (liuxuezhao) [reply src/client/api/event.c:527]: we probably can test if soak can pass if without this change?
just feel not quite straight forward why require the event cannot be set as ready by another thread, could it cause any potential problem?

Comment (mchaarawi) [reply src/client/api/event.c:527]: we tested without this change and it failed on 2.2
my theory is that two threads come in here;
main thread gets lock first, marks it as ready; other thread is in lock.
main thread continues to unlock and reinit (while other thread is still stuck in lock - being slow).
main thread reuses the event and makes the event in progress / running state.
other thread finally exits the lock and sees it's running and sets it as ready.

Comment (mchaarawi) [reply src/client/api/event.c:527]: yes i think i messed up with the checks there and mistaken it with the earlier check. you are correct. 
please ask Maureen to run 2.2 with only your PR and let's see.

**Review by liw — APPROVED** [review]

**Pull request closed without merging**